### PR TITLE
add non schema types to pkg/validation/spec

### PR DESCRIPTION
**Pull request by gautierdelorme**
part of https://github.com/kubernetes/kubernetes/issues/96303

Signed-off-by: Gautier Delorme <gautier.delorme@gmail.com>

### CODE REVIEW

File: pkg/builder/openapi.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 	"strings"
 
 	restful "github.com/emicklei/go-restful"
-	"github.com/go-openapi/spec"
+	"k8s.io/kube-openapi/pkg/validation/spec"
```
Comment (sttts) [inline pkg/builder/openapi.go:26]: order

File: pkg/validation/spec/header.go [hunk 2]
```diff
@@ -14,6 +14,84 @@
...
+		return err
+	}
+	if err := json.Unmarshal(data, &h.VendorExtensible); err != nil {
+		return err
+	}
+	return json.Unmarshal(data, &h.HeaderProps)
+}
+
+// JSONLookup look up a value by the json property name
+func (h Header) JSONLookup(token string) (interface{}, error) {
```
Comment (sttts) [inline pkg/validation/spec/header.go:76]: this this used?

**Comment by gautierdelorme** [comment]
https://github.com/kubernetes/kubernetes/pull/101234 tests look good

**Pull request merged**